src/rollup/utils/persist/expiring-local-storage.js

```diff
@@ -0,0 +1,58 @@
+var PAYLOAD_KEY = '$';
+var EXPIRES_KEY = '@';
+
+function safeJsonParse (json) {
+  try {
+    return JSON.parse(json);
+  } catch (e) {
+    // recover
+  }
+}
+
+export default {
+  getItem: function (key) {
+    var textValue = localStorage.getItem(key);
+    var value = safeJsonParse(textValue);
+
+    if (value && EXPIRES_KEY in value) {
+      if (Date.parse(value[EXPIRES_KEY]) < Date.now()) {
+        localStorage.removeItem(key);
+        return null;
+      }
+      if (PAYLOAD_KEY in value) {
+        // Extend the expiration date:
+        this.setItem(key, value[PAYLOAD_KEY]);
+        return value[PAYLOAD_KEY];
+      }
+    }
+    return textValue;
+  },
+
+  setItem: function (key, payload, expiresOn) {
+    if (!expiresOn) {
+      expiresOn = new Date();
+      expiresOn.setDate(expiresOn.getDate() + 365);
+    }
+
+    var record = {};
+
+    record[PAYLOAD_KEY] = '' + payload;
```

Comment:
why do we do `'' + payload` here? Just to ensure it's a string?

Reply:
yeah, this casts as a string to match the behavior of `localStorage`. We _could_ store any serializable value instead, but that's not what localStorage does 🤷‍♂